main.tf

```
@@ -8,13 +8,14 @@ module "ssh-key" {
 }
 
 resource "azurerm_kubernetes_cluster" "main" {
-  name                    = var.cluster_name == null ? "${var.prefix}-aks" : var.cluster_name
-  kubernetes_version      = var.kubernetes_version
-  location                = data.azurerm_resource_group.main.location
-  resource_group_name     = data.azurerm_resource_group.main.name
-  dns_prefix              = var.prefix
-  sku_tier                = var.sku_tier
-  private_cluster_enabled = var.private_cluster_enabled
+  name                            = var.cluster_name == null ? "${var.prefix}-aks" : var.cluster_name
+  kubernetes_version              = var.kubernetes_version
+  location                        = data.azurerm_resource_group.main.location
+  resource_group_name             = data.azurerm_resource_group.main.name
+  dns_prefix                      = var.prefix
+  sku_tier                        = var.sku_tier
+  private_cluster_enabled         = var.private_cluster_enabled
+  api_server_authorized_ip_ranges = length(var.api_server_authorized_ip_ranges) > 0 ? var.api_server_authorized_ip_ranges : []
```

Can be simplified to the below given the default value of `api_server_authorized_ip_ranges` is `[]`
```suggestion
  api_server_authorized_ip_ranges = var.api_server_authorized_ip_ranges
```